File: 10110.py
from math import sqrt
def f(n):
    A = [0 for _ in range(n+1)]
    for i in range(1, n+1):
        j = 1
        while i*j <= n:
            A[i*j] ^= 1
            j += 1
    # print(A[-1])
    return A[-1]

m = 5000
t = [i for i in range(1, m) if f(i) == 1]
s = [i**2 for i in range(1, int(sqrt(m)+1))]
print(t)
print(s)
print(t == s)

File: 10110.c
#include <math.h>
#include <stdint.h>
#include <stdio.h>

int main() {
    uint32_t n;
    while (scanf("%u", &n), n) {
        uint32_t x = (uint32_t) sqrt(n);
        if (x*x == n) {
            printf("yes\n");
        } else {
            printf("no\n");
        }
    }
    return 0;
}
